test/e2e/proxy_test.go

```diff
@@ -68,7 +70,493 @@ func skipIfProxyDisabled(t *testing.T) {
 func testProxyServiceSessionAffinityCase(t *testing.T, data *TestData) {
 	if len(clusterInfo.podV4NetworkCIDR) != 0 {
 		ipFamily := corev1.IPv4Protocol
-		testProxyServiceSessionAffinity(&ipFamily, []string{"169.254.169.253", "169.254.169.254"}, data, t)
+		testProxyServiceSessionAffinity(&ipFamily, []string{"169.254.169.1", "169.254.169.2"}, data, t)
+	}
+	if len(clusterInfo.podV6NetworkCIDR) != 0 {
+		ipFamily := corev1.IPv6Protocol
+		testProxyServiceSessionAffinity(&ipFamily, []string{"fd75::aabb:ccdd:ef00", "fd75::aabb:ccdd:ef01"}, data, t)
+	}
+}
+
+func skipIfProxyAllDisabled(t *testing.T, data *TestData) {
+	isProxyFull, err := data.IsProxyAll()
+	if err != nil {
+		t.Fatalf("Error getting option antreaProxy.proxyAll value")
+	}
+	if !isProxyFull {
+		t.Skipf("Skipping test because option antreaProxy.proxyAll is not enabled")
+	}
+}
+
+func skipIfKubeProxyEnabledOnLinux(t *testing.T, data *TestData, nodeName string) {
+	pods, err := data.clientset.CoreV1().Pods(antreaNamespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("Error fetching pods: %v", err)
+	}
+	for _, pod := range pods.Items {
+		if strings.Contains(pod.Name, "kube-proxy") && pod.Spec.NodeName == nodeName {
+			t.Skipf("Skipping test because kube-proxy is running")
+		}
+	}
+}
+
+func TestProxyLoadBalancerServiceIPv4(t *testing.T) {
+	testProxyLoadBalancerService(t, false)
+}
+
+func TestProxyLoadBalancerServiceIPv6(t *testing.T) {
+	skipIfNotIPv6Cluster(t)
+	testProxyLoadBalancerService(t, true)
+}
+
+func testProxyLoadBalancerService(t *testing.T, isIPv6 bool) {
+	skipIfProxyDisabled(t)
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	skipIfProxyAllDisabled(t, data)
+	skipIfHasWindowsNodes(t)
+	skipIfNumNodesLessThan(t, 2)
```

Comment:
Why they are not before setting up test?

Reply:
Thanks,  `skipIfHasWindowsNodes(t)` and `skipIfNumNodesLessThan(t, 2)` can be placed before setting up test, but `skipIfProxyAllDisabled` needs parameter `data`,  so it should be after setting up test.